### PR TITLE
Abstract parent table

### PR DESCRIPTION
**Pull request by dakota**
Fixes an issue with the model annotator not annotating models with an abstract parent class.

### CODE REVIEW

File: tests/test_files/Model/Table/BarBarsAbstractTable.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * @mixin \Tools\Model\Behavior\ConfirmableBehavior
+ * @mixin \MyNamespace\MyPlugin\Model\Behavior\MyBehavior
+ */
+class BarBarsAbstractTable extends AbstractTable {
+
+	/**
+	 * @param array $config
+	 * @return void
+	 */
+	public function initialize(array $config) {
```
Comment (dereuromark) [inline tests/test_files/Model/Table/BarBarsAbstractTable.php]: Proper indentation (tab) is off here.

Comment (dakota) [reply tests/test_files/Model/Table/BarBarsAbstractTable.php]: Not sure what exactly is wrong here?

Comment (dereuromark) [reply tests/test_files/Model/Table/BarBarsAbstractTable.php]: maybe github is confused.

but

...................................................S.......PHP Fatal error:  Cannot instantiate abstract class App\Model\Table\AbstractTable in /home/travis/build/dereuromark/cakephp-ide-helper/src/Annotator/ModelAnnotator.php on line 304

sounds like an actual issue.

**Comment by dereuromark** [comment]
Looks like this specific thing is only possible for PHP7+?
How could we prevent this inside PHP5.6? Checking for abstract class and then not trying to read it?

**Comment by dakota** [comment]
> How could we prevent this inside PHP5.6?

Not supporting outdated PHP versions :trollface:

Could use reflection to determine if it's an abstract class? Not sure your thoughts on reflection though.

**Comment by dereuromark** [comment]
Try it :) I am fine with whatever works.

**Comment by dereuromark** [comment]
Is it possible that it still runs into it?
https://travis-ci.org/dereuromark/cakephp-ide-helper/jobs/598090842#L393

**Comment by dakota** [comment]
It's failing because the normal flow of the annotator is still trying to instantiate the abstract class. Just pushed what should hopefully be the fix.

**Comment by dakota** [comment]
Still failing :( Gonna try to get PHP 5.6 locally and test locally.

**Comment by dereuromark** [comment]
AWESOME WORK :) THX MAN

**Pull request merged**